File: adv_2022/adv_day_06/cpp/day_06.cpp
#include <iostream>
#include <fstream>
#include <vector>   

int main(){
    std::ifstream File("input.txt");
    
    char data;
    int packet_start_position = 0;
    std::vector<char> character_sequences;
    
    while (File.good()) {
        File.get(data);
        character_sequences.push_back(data);
        packet_start_position++;
        bool start_position_flag = true;
        if (character_sequences.size() == 4) {
            for (int i = 0; i < 3; i++) {
                for (int j = i+1; j < 4; j++) {
                    if(character_sequences[i] == character_sequences[j]) {
                        start_position_flag = false;
                        character_sequences.erase(character_sequences.begin());
                        i = 4;
                        break;
                    }
                }
            }
            if (start_position_flag) {
                break;
            }
        }
    }

    File.close();
    std::cout << std::to_string(packet_start_position) << std::endl;
    // 1876
}

File: adv_2022/adv_day_06/python/day_06.py

with open("input.txt") as f:
    datastream = f.read()

packet_start_position = 0
character_sequences = []


for data_index, data in enumerate(datastream):
    character_sequences.append(data)
    start_position_flag = True
    if(len(character_sequences) == 4):
        for index in range(3):
            if(character_sequences.count(character_sequences[index]) > 1):
                start_position_flag = False
                character_sequences.pop(0)
                break
        
        if(start_position_flag):
            packet_start_position = data_index
            break

print(packet_start_position + 1)
# 1876